text/0374-jsii-modular.md

```diff
@@ -0,0 +1,256 @@
+> this release. Similar to the "working backwards", this section should be
+> written in a language as if the feature is now released.
+>
+> The template includes a some common questions, feel free to add any questions
+> that might be relevant to this feature or omit questions that you feel are not
+> applicable.
+
+### What are we launching today?
+
+We are announcing a change in versioning strategy for the *jsii* constellation
+of packages, which opens the way for an increased pace of innovation in the
+*jsii* ecosystem:
+
+- Each package of the constellation now has it's own, independent version
+  number, and only gets a new release when a change was introduced;
+- The `jsii` compiler now follows the `typescript`  compiler version number,
+  allowing developers to easily leverage improvements in the *TypeScript*
+  language and compiler performance;
+- The `@jsii/spec` package guarantees all releases of packages in the
+  constellation remain inter-operable going forward.
+
+### Why should I use this feature?
+
+The new versioning strategy allows developers to have greater control over the
+*jsii* tools they use. It allows developers to benefit from improvements to the
+*TypeScript* language (and compiler performance) faster, by allowing them to
+upgrade their project on their own timeline - in the same way they would be able
+to do for a "pure" *TypeScript* project.
+
+Since all parts of the *jsii* constellation are inter-operable with all major
+releases of the *jsii* packages, developers are able to opt into new and
+improved code-generation features of `jsii-pacmak` on their own schedule (i.e:
+when they release a new major version of their library) without requiring all
+their dependencies and dependents use the same version of `jsii-pacmak`.
+
+## Internal FAQ
+
+### Why are we doing this?
+
+We are doing this to reduce the friction involved in making major changes to any
+of the *jsii* constellation of packages. This will allow breaking changes to be
+introduced in parts of the ecosystem without bumping the major version of the
+rest of packages.
+
+This will improve our pace of innovation, as we will hence be able to introduce
+a new major version of any part of the ecosystem wihtout necessarily requiring
+all downstream consumers to migrate all at once.
+
+For example, this allows the `jsii` compiler to be upgraded to a new release
+line of `typescript` without forcing every downstream consumer to also accept
+the syntax breaking changes at the same time. Instead, consumers will be free
+to take on a new release line of `jsii` independently from their dependencies as
+well as from their consumers.
+
+Separating the mono-repository into multiple repositories will help ensure we
+maintain backwards compatibility between individual packages, and reduce the
+risk of creating codependent packages (different repositories need to be
+released independenlty, so dependencies can only flow in an asyclic manner).
```

Comment:
Don't monorepos provide the same or some of these assurances? For eg, I believe you can't have cyclic dependencies within packages of a monorepo.

It's not clear to me (from the rfc) why coupling jsii versions and breaking off monorepo are related. 

By not breaking off the monorepo, some of the concerns you've listed in the next question can be avoided while still decoupling jsii release versions.